### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by adamconder**
- Refactor controller into individual methods (separation of concerns)
- Refactoring logs to highlight switches enabled

### CODE REVIEW

File: app/uk/gov/hmrc/brm/filters/Filters.scala [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    * @param payload request transformed into Payload
+    * @return Tuple of List[Filter], list of failed filters
+    */
+  def process(payload : Payload) : List[Filter] = {
+
+    @tailrec
+    def filterHelper(uncheckedFilters : List[Filter], failedFilters : List[Filter]) : List[Filter] = {
+      if (failedFilters.nonEmpty && uncheckedFilters.isEmpty) {
+        BRMLogger.info("Filters", "process", s"Stopping due to failing a Filter, " +
+          s"remaining: $uncheckedFilters, failed: $failedFilters")
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/filters/Filters.scala]: Unchecked will always be a Nil list now.

File: app/uk/gov/hmrc/brm/filters/Filters.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import uk.gov.hmrc.brm.models.brm.Payload
+import uk.gov.hmrc.brm.utils.{BRMLogger, BirthRegisterCountry}
+
+import scala.annotation.tailrec
+
+/**
+  * Created by mew on 12/05/2017.
+  */
+
+trait FilterResults {
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/filters/Filters.scala]: These types can now be removed.

File: app/uk/gov/hmrc/brm/filters/Filters.scala [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    val baseWithFilters = baseFilters ::: filters.filter(f => shouldProcessFilter(f, payload))
+
+    BRMLogger.info("Filters", "getFilters", s"processing the following filters: $baseWithFilters")
+
+    baseWithFilters
+  }
+
+  /**
+    * @param payload request transformed into Payload
+    * @return Tuple of List[Filter], list of failed filters
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/filters/Filters.scala]: Outdated.

File: test/uk/gov/hmrc/brm/config/FeatureSwitchSpec.scala [hunk 4]
```diff
@@ -265,55 +264,55 @@ class FeatureSwitchSpec extends UnitSpec with BeforeAndAfter with OneAppPerTest
...
         "return false when parent feature is enabled but detail feature is disabled" taggedAs Tag("detailfeaturedisabled") in {
           implicit val payload: Payload = buildPayload(None, BirthRegisterCountry.SCOTLAND)
-          FeatureFactory().validate() shouldBe false
+          DownstreamFeatureFactory().enabled() shouldBe false
         }
 
         "return true when parent feature is enabled and detail feature is enabled" taggedAs Tag("detailfeatureenabled") in {
           implicit val payload: Payload = buildPayload(None, BirthRegisterCountry.SCOTLAND)
-          FeatureFactory().validate() shouldBe true
+          DownstreamFeatureFactory().enabled() shouldBe true
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/config/FeatureSwitchSpec.scala]: This spec can be removed?

Comment (chris-ianson) [reply test/uk/gov/hmrc/brm/config/FeatureSwitchSpec.scala]: @adamconder Yeh, this one isn't used it seems (you've moved/updated it in config/ now, right?)

Comment (chris-ianson) [reply test/uk/gov/hmrc/brm/config/FeatureSwitchSpec.scala]: @adamconder DateOfBirthFilter requires FilterResults?!

Comment (chris-ianson) [reply test/uk/gov/hmrc/brm/config/FeatureSwitchSpec.scala]: @adamconder ^^ I've updated that

**Pull request merged**